### ISSUE TITLE
Undefined method `length' for 138213544:Integer

### ISSUE THREAD

**Issue by johnathanludwig** [librato/librato-rails#140]
Since upgrading to ruby 2.4, we have been getting this error randomly from librato-rack/rails. It only started occurring after the ruby upgrade and is occurring in regular requests and tests though very infrequently.

I couldn't determine what the issue was with the code. Any ideas?

### Dependencies

- Ruby 2.4
- Rails 5.0.2
- librato-metrics (2.1.0)
- librato-rack (2.0.1)
- librato-rails (2.1.0)

### Stacktrace
```
vendor/bundle/ruby/2.4.0/gems/librato-rack-2.0.1/lib/librato/collector/aggregator.rb:92:in `measure': undefined method `length' for 138213544:Integer (NoMethodError)
from vendor/bundle/ruby/2.4.0/gems/librato-rack-2.0.1/lib/librato/collector/group.rb:23:in `measure'
from vendor/bundle/ruby/2.4.0/gems/librato-rails-2.1.0/lib/librato/rails/subscribers/controller.rb:21:in `block (2 levels) in <module:Subscribers>'
from vendor/bundle/ruby/2.4.0/gems/librato-rack-2.0.1/lib/librato/collector.rb:38:in `group'
from vendor/bundle/ruby/2.4.0/gems/librato-rails-2.1.0/lib/librato/rails/subscribers/controller.rb:19:in `block in <module:Subscribers>'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/fanout.rb:127:in `finish'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/fanout.rb:46:in `block in finish'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/fanout.rb:46:in `each'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/fanout.rb:46:in `finish'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/instrumenter.rb:42:in `finish_with_state'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/instrumenter.rb:27:in `instrument'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications.rb:164:in `instrument'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_controller/metal/params_wrapper.rb:248:in `process_action'
from vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/abstract_controller/base.rb:126:in `process'
from vendor/bundle/ruby/2.4.0/gems/actionview-5.0.2/lib/action_view/rendering.rb:30:in `process'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_controller/metal.rb:190:in `dispatch'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_controller/metal.rb:262:in `dispatch'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/routing/route_set.rb:32:in `serve'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/journey/router.rb:39:in `block in serve'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/journey/router.rb:26:in `each'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/journey/router.rb:26:in `serve'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/routing/route_set.rb:725:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-attack-5.0.1/lib/rack/attack.rb:140:in `call'
from vendor/bundle/ruby/2.4.0/gems/warden-1.2.6/lib/warden/manager.rb:35:in `block in call'
from vendor/bundle/ruby/2.4.0/gems/warden-1.2.6/lib/warden/manager.rb:34:in `catch'
from vendor/bundle/ruby/2.4.0/gems/warden-1.2.6/lib/warden/manager.rb:34:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/etag.rb:25:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/conditional_get.rb:25:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/head.rb:12:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/session/abstract/id.rb:222:in `context'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/session/abstract/id.rb:216:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/cookies.rb:613:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/callbacks.rb:38:in `block in call'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/callbacks.rb:97:in `__run_callbacks__'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/callbacks.rb:750:in `_run_call_callbacks'
from vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/callbacks.rb:90:in `run_callbacks'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/callbacks.rb:36:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/debug_exceptions.rb:49:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
from vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/rack/logger.rb:36:in `call_app'
from vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/rack/logger.rb:24:in `block in call'
from vendor/bundle/ruby/2.4.0/gems/semantic_logger-3.4.1/lib/semantic_logger/semantic_logger.rb:318:in `tagged'
from vendor/bundle/ruby/2.4.0/gems/semantic_logger-3.4.1/lib/semantic_logger/base.rb:123:in `tagged'
from vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/rack/logger.rb:24:in `call'
from config/initializers/silence_logger.rb:20:in `call'
from vendor/bundle/ruby/2.4.0/gems/request_store-1.3.2/lib/request_store/middleware.rb:9:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/request_id.rb:24:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/method_override.rb:22:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/runtime.rb:22:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/executor.rb:12:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/static.rb:136:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/deflater.rb:35:in `call'
from vendor/bundle/ruby/2.4.0/gems/rack-2.0.1/lib/rack/sendfile.rb:111:in `call'
from vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.2/lib/action_dispatch/middleware/ssl.rb:84:in `call'
from config/initializers/extensions/action_dispatch/ssl.rb:8:in `call'
from vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/engine.rb:522:in `call'
from vendor/bundle/ruby/2.4.0/gems/puma-3.6.2/lib/puma/configuration.rb:225:in `call'
from vendor/bundle/ruby/2.4.0/gems/puma-3.6.2/lib/puma/server.rb:578:in `handle_request'
from vendor/bundle/ruby/2.4.0/gems/puma-3.6.2/lib/puma/server.rb:415:in `process_client'
from vendor/bundle/ruby/2.4.0/gems/puma-3.6.2/lib/puma/server.rb:275:in `block in run'
from vendor/bundle/ruby/2.4.0/gems/puma-3.6.2/lib/puma/thread_pool.rb:116:in `block in spawn_thread'
```

**Comment by nextmat** [comment]
This looks like it may be the same issue as https://github.com/librato/librato-rack/issues/70

Unfortunately delegation was broken in the stdlib in 2.4.0, but the issue only occurs on some calls so the behavior is intermittent.

The fix is to upgrade to 2.4.1

**Comment by johnathanludwig** [comment]
Ok thank you. I glanced at that one before and thought it was unrelated.
